File: pages/SearchPage.ts
import { BasePage } from '../helpers/BasePage'

const inpQuery = "input[name='q']"

export class SearchPage extends BasePage {

    async openSearchPage() {
        await this.page.goto("https://google.com/")
        return this
    }

    async typeQuery(query: string) {
        await this.page.type(inpQuery, query)
        return this
    }

    async submitSearch() {
        await this.page.press(inpQuery, "Enter")
        return this
    }
}

File: src/main/kotlin/pages/SearchPage.kt
package pages

import helpers.PlaywrightObject.page
import io.qameta.allure.Step

class SearchPage {
    private val inpQuery get() = page?.locator("textarea[name='q']")
    private val txtTitle get() = page?.locator("div[data-attrid='title']")

    @Step("Get query text")
    fun getQueryText(): String {
        return inpQuery!!.innerText()
    }

    @Step("Get title")
    fun getTitle(): String {
        return txtTitle!!.innerText()
    }
}